web/js/userTasksReport.js

```diff
@@ -285,12 +348,22 @@ Ext.onReady(function () {
             handler: findTasks,
         }],
     });
+    /* Allow listing of user tasks of other users only for an admin user */
+    if ( admin == "" ) {
+        Ext.getCmp('userLogin').setDisabled(true);
+    }
 
     /* Handler to invoke the search service */
     function findTasks () {
-                var baseParams = {
-                    'userId': userId,
-                };
+                var baseParams = {};
+
+                if (Ext.getCmp('userLogin').baseParams.userId == ""){
+                    // If no value was set in the combobox, default to the logged in user
+                    baseParams.userId = userId;
+                } else {
+                    baseParams.userId = Ext.getCmp('userLogin').baseParams.userId;
```

Comment:
I don't think you should be using the `baseParams` member of the combo box object to store the value. `baseParams` is intended to maintain a list of parameters that will be sent in every request to the server, and is a member of Store objects, not combo boxes.

It's better you use `Ext.getCmp('userLogin').getValue()` directly.